spec/locales-currencies-tz.html

```diff
@@ -219,15 +254,42 @@ <h1>AvailableNamedTimeZoneIdentifiers ( ): a List of Time Zone Identifier Record
-            1. NOTE: An implementation may need to resolve _identifier_ iteratively.
+          1. NOTE: The algorithm steps below are intended to correspond to the behaviour of <code>icu::TimeZone::getIanaID()</code> in the International Components for Unicode (<a href="https://icu.unicode.org/">ICU</a>) and the processes for maintaining time zone identifier data in the Unicode Common Locale Data Repository (<a href="https://cldr.unicode.org">CLDR</a>).
+          1. If _identifier_ is a Link name in the IANA Time Zone Database and _identifier_ is not present in the “TZ” column of <code>zone.tab</code> of the IANA Time Zone Database, then
+            1. Let _zone_ be the Zone name that _identifier_ resolves to, according to the rules for resolving Link names in the IANA Time Zone Database.
+            1. If _zone_ starts with *"Etc/"*, then
+              1. Set _primary_ to _zone_.
+            1. Else,
+              1. Let _identifierCountryCode_ be the <a href="https://www.iso.org/glossary-for-iso-3166.html">ISO 3166-1 Alpha-2</a> country code whose territory contains the geographical area corresponding to _identifier_.
+              1. Let _zoneCountryCode_ be the ISO 3166-1 Alpha-2 country code whose territory contains the geographical area corresponding to _zone_.
+              1. If _identifierCountryCode_ is _zoneCountryCode_, then
+                1. Set _primary_ to _zone_.
+              1. Else,
+                1. Let _countryCodeLineCount_ be the number of lines in file <code>zone.tab</code> of the IANA Time Zone Database where the “country-code” column is _identifierCountryCode_.
+                1. If _countryCodeLineCount_ is 1, then
+                  1. Let _countryCodeLine_ be the line in file <code>zone.tab</code> of the IANA Time Zone Database where the “country-code” column is _identifierCountryCode_.
+                  1. Set _primary_ to the contents of the “TZ” column of _countryCodeLine_.
+                1. Else,
+                  1. Let _backzone_ be *undefined*.
+                  1. Let _backzoneLinkLines_ be the List of lines in the file <code>backzone</code> of the IANA Time Zone Database that start with either *"Link "* or *"#PACKRATLIST zone.tab Link "*.
+                  1. For each element _line_ of _backzoneLinkLines_, do
+                    1. If _line_ starts with *"#PACKRATLIST zone.tab "*, set _line_ to the substring of _line_ from 22.
+                    1. Assert: _line_ starts with *"Link "*.
+                    1. Set _line_ to the substring of _line_ from 5.
+                    1. Let _backzoneAndLink_ be StringSplitToList(_line_, *" "*, 2).
+                    1. Assert: _backzoneAndLink_ has exactly two elements.
+                    1. If _backzoneAndLink_[1] is _identifier_, then
+                      1. Assert: _backzone_ is *undefined*.
+                      1. Set _primary_ to _backzoneAndLink_[0].
+                  1. Assert: _backzone_ is not *undefined*.
```

Comment:
```suggestion
                  1. Let _backzone_ be *undefined*.
                  1. Let _backzoneLinkLines_ be the List of lines in the file <code>backzone</code> of the IANA Time Zone Database that start with either *"Link "* or *"#PACKRATLIST zone.tab Link "*.
                  1. For each element _line_ of _backzoneLinkLines_, do
                    1. Let _i_ be StringIndexOf(_line_, *"Link"*, 0).
                    1. Set _line_ to the substring of _line_ from _i_.
                    1. Let _fields_ be StringSplitToList(_line_, *" "*).
                    1. Assert: _fields_ has at least three elements, _fields_[0] is *"Link"*, _fields_[1] is not the empty String, and _fields_[2] is not the empty String.
                    1. If _fields_[2] is _identifier_, then
                      1. Assert: _backzone_ is *undefined*.
                      1. Set _backzone_ to _fields_[1].
                  1. Assert: _backzone_ is not *undefined*.
                  1. Set _primary_ to _backzone_.
```

Reply:
Thanks for catching the bug!  I updated the spec text to take some of this suggestion but not all of it. Two things I didn't move over:
1. The `_fields_[0] is *"Link"*` assertion was redundant with the earlier StringIndexOf call (or would have been if the StringIndexOf call included the terminating space separator. So I started the StringSplitToList after `"Link "`.
2. I liked the clarity of `_backzoneAndLink_` as the local variable name instead of the more generic `_fields_`. This algorithm is pretty complex so I think any additional documentation is helpful here. 

This is in the latest commit. Let me know if this is OK!